scripts/set_assignees.py

```diff
@@ -93,30 +94,103 @@ def process_pr(gh, maintainer_file, number):
             continue
 
         all_areas.update(areas)
-        is_instance = False
+        #
+        # DJL: Don't understand what "is_instance" is trying to record?
+        # It mucks with the weighting though.
+        #
+        #is_instance = False
         sorted_areas = sorted(areas, key=lambda x: 'Platform' in x.name, reverse=True)
         for area in sorted_areas:
-            c = 1 if not is_instance else 0
+            #c = 1 if not is_instance else 0
+
+            area_counter[area] += 1
 
-            area_counter[area] += c
-            labels.update(area.labels)
             # FIXME: Here we count the same file multiple times if it exists in
             # multiple areas with same maintainer
             for area_maintainer in area.maintainers:
-                found_maintainers[area_maintainer] += c
+                found_maintainers[area_maintainer] += 1
 
-            if 'Platform' in area.name:
-                is_instance = True
+            #if 'Platform' in area.name:
+            #    is_instance = True
 
     area_counter = dict(sorted(area_counter.items(), key=lambda item: item[1], reverse=True))
+
+    # selectively pick some of the most likely labels
+    for item in area_counter:
+        prop = (area_counter[item] / num_files) * 100
+        #
+        # Not sure what the right percentage is. When a PR has touched
+        # a lot of files it will create a giant list. So try to find
+        # just the top areas touched and hopefully it will get the right
+        # labels.
+        #
+        # log(f"label {item.labels} : count {area_counter[item]} : {prop:.2f}%")
+        if prop > 15:
```

Comment:
it might be better to not even do this but to just ensure the highest priority labels are at the top of the list and then when assigning labels just clip to the limit we have below.